### PR TITLE
Remove pattern explorer modal

### PR DESCRIPTION
**Pull request by jeryj**
## What?

Removes the pattern explorer modal from the pattern inserter sidebar.

## Why?

The modal is not adding anything not already available in the pattern sidebar. Rather than have an alternate experience (that is not working well:
- Inserts patterns behind the modal (you can't see that they have been inserted)
- https://github.com/WordPress/gutenberg/issues/56442
- Tabbing around causes odd flashing and removal of content
- Grid layout but must use down arrow to navigate content

## How?

Remove it, rely on the pattern sidebar in the inserter

### CODE REVIEW

File: packages/block-editor/src/components/inserter/style.scss
Comment (stokesman) [inline packages/block-editor/src/components/inserter/style.scss:1]: Can’t lines 430-491 also be removed? The `.block-editor-block-patterns-explorer` stuff.

**Comment by jameskoster** [comment]
Iirc this modal used to be the only way to retrieve patterns from the directory. But since that's no longer the case it does feel a bit duplicative. I'd be interested to hear @richtabor's thoughts.

File: packages/block-editor/src/components/block-patterns-list/style.scss [hunk 1]
```diff
@@ -18,13 +18,6 @@
 
 .block-editor-block-patterns-list__item {
 	height: 100%;
-	// This is derived from the top padding set on
-	// `.block-editor-block-patterns-explorer__list`
-	scroll-margin-top: $grid-unit-30;
-	// This is derived from the bottom padding set on
-	// `.block-editor-block-patterns-explorer__list` and
-	// the bottom margin set on `...__list-item` above
-	scroll-margin-bottom: ($grid-unit-40 + $grid-unit-30);
```
Comment (jeryj) [inline packages/block-editor/src/components/block-patterns-list/style.scss:27]: I couldn't get these to apply when viewing the query pattern explorer or the sidebar inserter pattern explorer. None of them had a snapping effect for me.

File: packages/block-editor/src/components/inserter/style.scss
Comment (jeryj) [reply packages/block-editor/src/components/inserter/style.scss:1]: Good catch. Thanks!

**Comment by richtabor** [comment]
We shouldn't remove this, unless it's clear that people don't use it, and we replace any loss in functionality. 

As @jameskoster mentioned, it's how search results from the pattern directory can be pulled in.

**Comment by jeryj** [comment]
@richtabor What features are available in the modal that aren't available in the sidebar inserter?

**Comment by stokesman** [comment]
> it's how search results from the pattern directory can be pulled in.

From what I see the default "All" view is pulling in all the same patterns as when I open the explorer. That agrees with James’s comment https://github.com/WordPress/gutenberg/pull/63636#issuecomment-2232917523. There doesn’t seem to be an absolute need for the explorer anymore.

**Comment by jeryj** [comment]
Closing as it seems like we're not ready to remove this yet.

**Pull request closed without merging**